reaskapi/api_client.py

```diff
@@ -3,20 +3,28 @@
 import sys
 import time
 import requests
+from dataclasses import dataclass 
 from reaskapi.auth import get_access_token
 
 URL_MAX_BYTES = 2**15
 
+@dataclass
+class ClientConfig:
+    """Config class to customize ApiClient
+    """
+    base_url: str
+
+
 class ApiClient:
     logger = logging.getLogger(__name__)
 
-    def __init__(self, product, config_section='default'):
+    def __init__(self, product, config_section='default', config: ClientConfig=None):
         """
         Initialize client getting access token
         """
         self.access_token = get_access_token(config_section)
         self.product = product
-        self.base_url = 'https://api.reask.earth/v2'
+        self.base_url = config.base_url if config is not None else 'https://api.reask.earth/v2'
```

Comment:
Can you remove the 'config_section' argument and make it part of the ClientConfig? This needs to be done in the metryc.py and deepcyc.py files as well.

Reply:
I can put `config_section` into `ClientConfig`.
Removing `config_section` from the constructor breaks backward compatibility though. I can check the usage of the parameter in repos which uses `api` repo and update them.
Is there any repos using the reaskapi client other than `api-server` repo?

Reply:
OK, I've created an issue to do this https://github.com/reaskearth/api/issues/15

Reply:
I would put `config_section` into the `ClientConfig` and create a new constructor which takes `ClientConfig` only. I leave the original constructor for backward compatibility but mark it as 'deprecated'.
We can use the issue for removing the deprecated constructor.